From cf70a1872c5657ad0d8c498922f2a6e5b65c4c0d Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Wed, 9 Jul 2014 15:03:44 +0300
Subject: [PATCH 044/538] fix: xor: added shutdown capability to disable clocks

Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/9097
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/9142
Signed-off-by: Neta Zur <neta@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/16147

Conflicts:
	drivers/dma/mv_xor.c

Change-Id: I476be491dfffc1eff5a5a90dee0e617b4c0f9a0b
Reviewed-on: http://vgitil04.il.marvell.com:8080/27274
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
---
 drivers/dma/mv_xor.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/dma/mv_xor.c b/drivers/dma/mv_xor.c
index 2b1989a..b250f63 100644
--- a/drivers/dma/mv_xor.c
+++ b/drivers/dma/mv_xor.c
@@ -1295,10 +1295,21 @@ err_channel_add:
 	return ret;
 }
 
+static void mv_xor_shutdown(struct platform_device *pdev)
+{
+	struct mv_xor_device *xordev = platform_get_drvdata(pdev);
+
+	if (!IS_ERR(xordev->clk)) {
+		clk_disable_unprepare(xordev->clk);
+		clk_put(xordev->clk);
+	}
+}
+
 static struct platform_driver mv_xor_driver = {
 	.probe		= mv_xor_probe,
 	.suspend        = mv_xor_suspend,
 	.resume         = mv_xor_resume,
+	.shutdown	= mv_xor_shutdown,
 	.driver		= {
 		.name	        = MV_XOR_NAME,
 		.of_match_table = of_match_ptr(mv_xor_dt_ids),
-- 
1.9.1

